src/View.php

```diff
@@ -584,6 +583,83 @@ public function removeAttr($property)
 
     // }}}
 
+    // {{{ Sticky URLs
+
+    /** @var string[] stickyGet arguments */
+    public $stickyArgs = [];
+
+    /**
+     * Build an URL which this view can use for js call-backs. It should
+     * be guaranteed that requesting returned URL would at some point call
+     * $this->invokeInit().
+     *
+     * @param array $page
+     *
+     * @return string
+     */
+    public function jsUrl($page = [], bool $includeRunningCallback = false)
+    {
+        return $this->getApp()->jsUrl($page, false, array_merge($this->_getStickyArgs(), $includeRunningCallback ? $this->getCallbackArguments() : []));
+    }
+
+    /**
+     * Build an URL which this view can use for call-backs. It should
+     * be guaranteed that requesting returned URL would at some point call
+     * $this->invokeInit().
+     *
+     * @param string|array $page URL as string or array with page name as first element and other GET arguments
+     *
+     * @return string
+     */
+    public function url($page = [], bool $includeRunningCallback = false)
```

Comment:
Can the 2nd arg be an instance of the running Callback?

If yes, it will be more robust as we will not have to filter `$_GET` which can be spoofed.